tests/test_related.py

```
@@ -0,0 +1,521 @@
+        current_title = "Daily Devlog — Jan 15, 2025"
+        
+        # Mock the remote posts fetching to fail
+        with patch.object(related_service, '_fetch_published_posts_from_remote') as mock_remote:
+            mock_remote.side_effect = Exception("API Error")
+            
+            related_posts = related_service.find_related_posts(
+                current_date, current_tags, current_title, repo="test/repo"
+            )
+            
+            # Should still return local posts
+            assert len(related_posts) > 0
```

Ruff (0.12.2)

Use of `assert` detected

(S101)